Doc/deprecations/pending-removal-in-3.16.rst

```diff
@@ -8,6 +8,10 @@ Pending Removal in Python 3.16
 * :mod:`builtins`:
   ``~bool``, bitwise inversion on bool.
 
+* :mod:`socket`:
+  If type annotations are required, use :class:`~socket.socket` instead.
```

Comment:
You need to specify what got deprecated here.

Reply:
In addition, I would remove `~` because a fully-qualified name is better here.